py/serve.py

```diff
@@ -700,7 +700,16 @@ def resolve_tokens(proto):
     if 'ListFields' in dir(proto):
         for descriptor, message in proto.ListFields():
             if descriptor.name == 'bytes_value':
-                token = message[:20].decode('utf8')
+                try:
+                    token = message[:20].decode('utf8')
+                except UnicodeDecodeError:
+                    # Client generated tokens are utf-8 decodable.
+                    # (see js/uploads.js) So if the value is not utf-8
+                    # decodable, it's not an upload token, and we should proceed
+                    # to the next proto field. (Likely, the value is
+                    # actual binary data from an upload that's already been
+                    # processed.)
+                    pass
```

Comment:
I think you want `continue` here; `pass` will probably run into errors later on when you try to manipulate the token?

Reply:
Whoops, that should definitely be continue, I blanked out making that last change... Thanks for catching again.